infrastructure/templates/public-api/main.bicep

```diff
@@ -1,290 +1,206 @@
     containerAppImageName: containerAppImageName
     containerAppTargetPort: containerAppTargetPort
     useDummyImage: useDummyImage
-    dbConnectionString: eesKeyVault.getSecret(databaseModule.outputs.connectionStringSecretName)
-    serviceBusConnectionString: eesKeyVault.getSecret(serviceBusFunctionQueueModule.outputs.connectionStringSecretName)
+    dbConnectionString: keyVault.getSecret(postgreSqlServerModule.outputs.connectionStringSecretName)
     tagValues: tagValues
+    applicationInsightsKey: applicationInsightsModule.outputs.applicationInsightsKey
+    subnetId: vNetModule.outputs.apiContainerAppSubnetRef
   }
   dependsOn: [
-    keyVaultModule
-    databaseModule
-    serviceBusFunctionQueueModule
+    postgreSqlServerModule
+    applicationInsightsModule
+    vNetModule
   ]
 }
 
-//Deploy Service Bus
-module serviceBusFunctionQueueModule 'components/serviceBusQueue.bicep' = {
-  name: 'serviceBusQueueDeploy'
+// Deploy data-processing Function.
+module dataProcessorFunctionAppModule 'application/dataProcessorFunctionApp.bicep' = {
+  name: 'functionAppDeploy-data-processor'
   params: {
     resourcePrefix: resourcePrefix
     location: location
-    namespaceName: namespaceName
-    queueName:queueName
-    keyVaultName: keyVaultModule.outputs.keyVaultName
+    functionAppName: 'data-processor'
+    storageAccountConnectionString: storageAccountConnectionString
+    dbConnectionString: keyVault.getSecret(postgreSqlServerModule.outputs.connectionStringSecretName)
     tagValues: tagValues
+    applicationInsightsKey: applicationInsightsModule.outputs.applicationInsightsKey
+    subnetId: vNetModule.outputs.dataProcessorSubnetRef
   }
   dependsOn: [
-    keyVaultModule
+    postgreSqlServerModule
+    applicationInsightsModule
+    vNetModule
   ]
 }
 
-//Deploy ETL Function
-module etlFunctionAppModule 'application/processorFunctionApp.bicep' = {
-  name: 'processorFunctionAppDeploy'
-  params: {
-    resourcePrefix: resourcePrefix
-    location: location
-    functionAppName: functionAppName
-    storageAccountConnectionString: eesKeyVault.getSecret(storageAccountModule.outputs.connectionStringSecretName)
-    dbConnectionString: eesKeyVault.getSecret(databaseModule.outputs.connectionStringSecretName)
-    serviceBusConnectionString: eesKeyVault.getSecret(serviceBusFunctionQueueModule.outputs.connectionStringSecretName)
-    tagValues: tagValues
-  }
-  dependsOn: [
-    keyVaultModule
-    databaseModule
-    storageAccountModule
-    serviceBusFunctionQueueModule
-  ]
-}
-
-
-//outputs
-output containerRegistryLoginServer string = containerRegistryModule.outputs.containerRegistryLoginServer
-output containerRegistryName string = containerRegistryModule.name
-output metadataDatabaseRef string = databaseModule.outputs.databaseRef
+// Outputs.
+output containerRegistryLoginServer string = containerRegistry.properties.loginServer
+output containerRegistryName string = containerRegistry.name
```

Comment:
Currently no, and in fact currently we're not using any of the outputs of main.bicep at all.

I'm assuming that the main point of having these outputs from the Infra deploy job is so that you can then go on and use them as inputs in subsequent stages of the pipeline, and as we don't currently have any subsequent stages to carry on to, we don't need these outputs at all.

Given that, I've gone ahead and removed them all for now, but we can add them in later when they're needed (if they're needed!).